src/sentry/api/endpoints/team_members.py

```diff
@@ -3,23 +3,54 @@
 from rest_framework.response import Response
 
 from sentry.api.bases.team import TeamEndpoint
-from sentry.api.serializers import serialize
-from sentry.models import InviteStatus, OrganizationMember
+from sentry.api.serializers import Serializer, register, serialize
+from sentry.models import InviteStatus
+from sentry.models.organizationmemberteam import OrganizationMemberTeam
+
+
+@register(OrganizationMemberTeam)
+class DetailedOrganizationMemberTeamSerializer(Serializer):
+    def __init__(self, *args, **kwargs):
+        self.team = kwargs.pop("team", None)
+        super().__init__(*args, **kwargs)
+
+    def get_attrs(self, item_list, user, **kwargs):
+        attrs = {}
+        for org_member_team in item_list:
+            attrs[org_member_team] = {
+                "org_member": org_member_team.organizationmember,
+                "user": org_member_team.organizationmember.user,
+            }
+        return attrs
+
+    def serialize(self, obj, attrs, user):
+        org_member = serialize(attrs["org_member"])
+        org_member["user"] = serialize(attrs["user"])
+        org_member["teamRole"] = obj.role
+        org_member["teamSlug"] = self.team.slug
+        return org_member
 
 
 class TeamMembersEndpoint(TeamEndpoint):
     def get(self, request: Request, team) -> Response:
-        queryset = OrganizationMember.objects.filter(
-            Q(user__is_active=True) | Q(user__isnull=True),
-            organization=team.organization,
-            invite_status=InviteStatus.APPROVED.value,
-            teams=team,
-        ).select_related("user")
-
-        member_list = sorted(
-            queryset, key=lambda x: x.user.get_display_name() if x.user_id else x.email
+        queryset = (
+            OrganizationMemberTeam.objects.filter(
```

Comment:
Why did we switch this query to be on the `OrganizationMemberTeam` rather than the `OrganizationMember`? Will this now return multiple rows for users that exist on multiple teams?

Reply:
Actually I guess that since this is filtered to a specific team we'll be fine here.